### DISCUSSION TITLE
How to add Assets as Input for a Job/Graph

### DISCUSSION THREAD

**Discussion by PascalStehling** [dagster-io/dagster#10802] [Q&A] [answered]
Hi all,

is there a way to have Assets as input for a "normal" Job. I have some Assets and based on them I want to run a Job, which does not create a new Asset. For the job I want to reuse the already existing structure with the Asset and its IO Manager and not load the Data directly from the source DB. Is this possible? And if so, how? 

Below is an example of what i am thinking about. Its not working as the AssetIn's are directly passed as Input and not the loaded Asset values.

Thank you for your help and keep on the good work :)

```py
from dagster import asset, repository, job, op, OpExecutionContext, AssetIn, fs_io_manager


@asset(io_manager_def=fs_io_manager)
def get_five() -> int:
    return 5


@asset(io_manager_def=fs_io_manager)
def get_ten() -> int:
    return 10


@op
def do_something_special_op(context: OpExecutionContext, five: int,
                            ten: int) -> None:
    context.log.info(five + ten)


@job(resource_defs={"io_manager": fs_io_manager},
     input_values={
         "get_five": AssetIn("get_five", input_manager_key="io_manager"),
         "get_ten": AssetIn("get_ten", input_manager_key="io_manager"),
     })
def asset_based_job(get_five: int, get_ten: int) -> None:
    do_something_special_op(get_five, get_ten)


@repository
def repo() -> list:
    return [get_five, get_ten, asset_based_job]
```

**Comment by dpeng817** [comment]
we don't have a way of mixing assets in with regular ops at the moment. What you could do is have an input manager at the source of the job that reads from the location that the relevant asset(s) wrote to.

**Reply by PascalStehling** [reply]
Thanks for the answer. 
Is a feature like this planned? Maybe a function called `load_asset` which can be used in the input_values of a job?

```py
@job(resource_defs={"io_manager": fs_io_manager},
     input_values={
         "get_five": load_asset("get_five"),
         "get_ten": load_asset("get_ten"),
     })
def asset_based_job(get_five: int, get_ten: int) -> None:
    do_something_special_op(get_five, get_ten)

```

**Reply by sryza** [reply]
@PascalStehling I added an issue to track adding this: https://github.com/dagster-io/dagster/issues/10874. I can't promise that we'll get to it in the near future, because of other stuff we're working on, but I agree that this would be beneficial to add.

**Comment by yachal-upson** [comment]
For what we're working on, it also feels intuitive to want to execute a job based on materialised asset data. Thanks @PascalStehling for raising this.

**Reply by sryza** [reply]
@yachal-upson - take a look at this comment: https://github.com/dagster-io/dagster/discussions/10802#discussioncomment-5773567

**Comment by sryza** [comment, marked as answer]
This is now possible: https://docs.dagster.io/concepts/ops-jobs-graphs/graphs#loading-an-asset-as-an-input

In particular, you can write a job that uses assets as inputs.

It's not yet possible to have a single job materialize assets and then execute non-asset ops after. To accomplish that, you would need to use a multi-asset sensor to launch the non-asset job after the assets are materialized.

**Reply by yachal-upson** [reply]
Thanks for the pointer
